File: server/src/models/course.ts
import { Document, model, Schema, Types } from "mongoose";

export interface ICourse extends Document {
    _id: string
    title: string
    number: string
    credits: string
    offered: string
    contents: string
    driveFolder: string,
    driveFiles: Array<string>
    reviews: Array<Types.ObjectId>
    author: string
    dept: string
}

const CourseSchema = new Schema({
    title: { type: String, required: true, unique: true },
    number: { type: String, required: true, unique: true },
    credits: { type: String, required: true },
    offered: { type: String, required: true },
    contents: { type: String, required: true },
    driveFolder: { type: String, required: true },
    driveFiles: [{ type: String, required: true }],
    reviews: [{ type: Types.ObjectId, required: true, ref: "Review" }],
    author: { type: String, required: true },
    dept: { type: String, required: true }
})

export default model<ICourse>("Course", CourseSchema)

File: server-go/models/course.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

// Course Structure
type Course struct {
	_id         string               `bson:"_id,omitempty"`
	title       string               `bson:"title,omitempty"`
	number      string               `bson:"number,omitempty"`
	credits     string               `bson:"credits,omitempty"`
	offered     string               `bson:"offered,omitempty"`
	contents    string               `bson:"contents,omitempty"`
	driveFolder string               `bson:"driveFolder,omitempty"`
	driveFiles  []string             `bson:"driveFiles,omitempty"`
	reviews     []primitive.ObjectID `bson:"reviews,omitempty"`
	author      string               `bson:"author,omitempty"`
	dept        string               `bson:"dept,omitempty"`
}
